src/main/java/com/hierynomus/smbj/connection/Connection.java

```diff
@@ -71,21 +72,36 @@
     private static final Logger logger = LoggerFactory.getLogger(Connection.class);
     private ConnectionInfo connectionInfo;
 
+    private SMBClient client;
+    public SMBClient getClient() {
+        return client;
+    }
+
     private Config config;
     private TransportLayer transport;
     private final SMBEventBus bus;
     private PacketReader packetReader;
     private Thread packetReaderThread;
     private final ReentrantLock lock = new ReentrantLock();
 
-    public Connection(Config config, TransportLayer transport, SMBEventBus bus) {
+    public Connection(Config config, SMBClient client, TransportLayer transport, SMBEventBus bus) {
```

Comment:
Can we do this differently than linking the connection to the client? Now we have a cyclic package dependnecy.